p2p/protocol/identify/id.go

```diff
@@ -186,9 +186,11 @@ func (ids *IDService) populateMessage(mes *pb.Identify, c inet.Conn) {
 
 	// set our public key
 	ownKey := ids.Host.Peerstore().PubKey(ids.Host.ID())
-	if ownKey == nil {
+	prKey := ids.Host.Peerstore().PrivKey(ids.Host.ID())
```

Comment:
Let's make it clear *why* we're checking for the private key. Something like:

```go
if ownKey == nil {
  // check if we're even operating in "secure mode"
  if ids.Host.Peerstore().PrivKey(ids.Host.ID()) != nil {
    // warn
  }
} else ...
```

(this also avoid storing the private key to make it clear that we're *just* checking to see if we have it)